io/include/pcl/io/ply/ply_parser.h

```diff
@@ -559,9 +559,16 @@ inline bool pcl::io::ply::ply_parser::parse_scalar_property (format_type format,
   typedef ScalarType scalar_type;
   if (format == ascii_format)
   {
+    std::string value_s;
     scalar_type value = std::numeric_limits<scalar_type>::quiet_NaN ();
     char space = ' ';
-    istream >> value;
+    istream >> value_s;
+    try
+    {
+      value = boost::lexical_cast<scalar_type> (value_s);
+    }
+    catch (boost::bad_lexical_cast &) {}
```

Comment:
I have the impression that ```boost::lexical_cast```only throws ```boost::bad_lexical_cast``` exceptions, but I can extend it for std::exception. 

If the conversion fails ```value``` maintains its original value, in this case ```std::numeric_limits<scalar_type>::quiet_NaN```. [Here's](http://en.cppreference.com/w/cpp/types/numeric_limits/quiet_NaN) the actual value based on each type, for reference.